### PR TITLE
Parse backend requests for valid inputs

### PR DESCRIPTION
**Pull request by huy-vuong**
This uses [`yup`](https://github.com/jquense/yup) to parse inputs and verify that they're valid for the backend's `GET /rsvps/:lastName/:firstName` endpoint.

### CODE REVIEW

File: api/test/parse.test.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+   test('returns the first and last name trimmed if valid', async () => {
+      expect(
+         await parseName({ firstName: ' f ', lastName: '  l ' })
+      ).toMatchObject({ firstName: 'f', lastName: 'l' });
+   });
+   test('throws a ValidationError if the first name is absent', async () => {
+      expect(parseName({ lastName: 'lahst' })).rejects.toThrow(ValidationError);
+      expect(parseName({ firstName: null, lastName: 'lahst' })).rejects.toThrow(
+         ValidationError
+      );
```
Comment (YanZ777) [inline api/test/parse.test.js:14]: **question:** Why not split these up into two tests?

File: api/test/parse.test.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
+const { ValidationError } = require('yup');
+const { parseName } = require('../src/parse');
+
+describe('parseName', () => {
+   test('returns the first and last name trimmed if valid', async () => {
```
Comment (YanZ777) [inline api/test/parse.test.js:5]: **question:** Would these be better off as being table driven tests?

**Review by YanZ777 — APPROVED** [review]

File: api/test/parse.test.js (hunk 2 above)
Comment (YanZ777) [reply api/test/parse.test.js:5]: **info:** Here's the [jest documentation on how to write table driven tests](https://jestjs.io/docs/api#testeachtablename-fn-timeout).

File: api/test/parse.test.js (hunk 1 above)
Comment (huy-vuong) [reply api/test/parse.test.js:14]: Yeah, that's a fair point. I can do that right now.

File: api/test/parse.test.js (hunk 2 above)
Comment (huy-vuong) [reply api/test/parse.test.js:5]: Oh neat, thanks for showing me this. It's nice to use in conjunction with `describe()` to better nest the tests to read like a spec.

**Pull request merged**